src/sentry/models/releasefile.py

```diff
@@ -59,3 +60,25 @@ def get_ident(cls, name, dist=None):
         if dist is not None:
             return sha1_text(name + '\x00\x00' + dist).hexdigest()
         return sha1_text(name).hexdigest()
+
+    @classmethod
+    def normalize(cls, url):
+        """Transforms a full absolute url into 2 or 4 generalized options
+
+        * the original url as input
+        * (optional) original url without querystring
+        * the full url, but stripped of scheme and netloc
+        * (optional) full url without scheme and netloc or querystring
+        """
+        # Always ignore the fragment
+        scheme, netloc, path, query, _ = urlsplit(url)
```

Comment:
This is all a tad bit verbose, but it's a bunch of awkward parts, so chose to be overly explicit for sake of clarity.